site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,507 @@
+object. The keywords in storage will populate the suggestions. When user enters the omnibox keyword
+(api) in the address bar followed by tab or space, Chrome will display a list of suggestions.
+
+Now, let's register the omnibox event listeners at the top level of the script. When the user enters the omnibox keyword "api" in the address bar followed by tab or space, Chrome will display a list of suggestions based on the keywords in storage. The [onInputChanged()][omnibox-input-changed] event, which takes the current user input and a suggestResult object, is responsible for populating these suggestions.
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+const URL_CHROME_EXTENSIONS_DOC =
+  'https://developer.chrome.com/docs/extensions/reference/';
+const NUMBER_OF_PREVIOUS_SEARCHES = 4;
+
+// Display the suggestions after user starts typing
+chrome.omnibox.onInputChanged.addListener(async (input, suggest) => {
+  await chrome.omnibox.setDefaultSuggestion({
+    description: 'Enter a Chrome API or choose from past searches'
+  });
+  const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
+  const suggestions = apiSuggestions.map((api) => {
+    return { content: api, description: `Open chrome.${api} API` };
+  });
+  suggest(suggestions);
+});
+```
+
+After the user selects a suggestion, [`onInputEntered()`][omnibox-input-entered] will open the corresponding Chrome API reference page.
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+// Open the reference page of the chosen API
+chrome.omnibox.onInputEntered.addListener((input) => {
+  chrome.tabs.create({ url: URL_CHROME_EXTENSIONS_DOC + input });
+  // Save the latest keyword
+  updateHistory(input);
+});
+```
+
+The `updateHistory()` function below takes the omnibox input and saves it to [storage.local][api-storage]. This way the most recent search term can be used later as an omnibox suggestion.
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+async function updateHistory(input) {
+  const { apiSuggestions } = await chrome.storage.local.get('apiSuggestions');
+  apiSuggestions.unshift(input);
+  apiSuggestions.splice(NUMBER_OF_PREVIOUS_SEARCHES);
+  return chrome.storage.local.set({ apiSuggestions });
+}
+```
+
+{% Aside 'important' %}
+
+Extension service workers can use both web APIs and Chrome APIs, with a few exceptions. For more information, see [Service Workers events](tbd).
+
+{% endAside %}
+
+### Step 6: Set up a recurring event {: #step-6 }
+
+The `setTimeout()` or `setInterval()` methods are commonly used to perform delayed or periodic
+tasks. However, these APIs can fail because the scheduler will cancel the timers when the service
+worker is terminated. Instead, extensions can use the [`chrome.alarms`][api-alarms] API. 
+
+Start by requesting the `"alarms"` permission in the manifest. Additionally, to fetch the extension tips from a remote hosted location, you need to request [host permission][doc-host-perm]:
```

Comment:
```suggestion
Start by requesting the `alarms` permission in the manifest. Additionally, to fetch the extension tips from a remote hosted location, you need to request [host permission][doc-host-perm]:
```

Reply:
This is the convention we've been following for permissions and manifest keys